jdaviz/core/template_mixin.py

```diff
@@ -397,15 +402,24 @@ def __init__(self, plugin, items, selected, selected_has_subregions=None,
                            handler=lambda msg: self._delete_subset(msg.subset))
 
         # intialize any subsets that have already been created
-        for lyr in self.app.data_collection.subset_groups:
-            self._update_subset(lyr)
+        self._reset_items()
+
+    def _reset_items(self):
+        all_subsets = [self._subset_to_dict(s) for s in
+                       self.app.data_collection.subset_groups]
+        self.items = self.manual_items + [s for s in all_subsets
+                                          if self._allowed_type is None
+                                          or s['type'] == self._allowed_type]
 
     @staticmethod
     def _subset_type(subset):
         if isinstance(subset.subset_state, RoiSubsetState):
-            # then this is a spatial subset, we want to ignore
+            # then this is a spatial subset,
             return 'spatial'
         else:
+            # TODO: more advanced logic.  This is counting static regions in imviz as spectral
+            # (which isn't an issue so long as those are isolated to imviz which will never use
+            # subset selects with spectral regions)
```

Comment:
What if one day, Cubeviz decides to use Imviz viewer? 😬

Reply:
then we'll need the more advanced logic (I'd probably suggest modifying the if-statement to catch the static regions as spatial and then implementing additional filters to be able to exclude them for aperture photometry).  We'll also need to do that if any plugins in Imviz support static regions and want them included in the dropdowns.